src/agent/src/netlink.rs

```diff
@@ -422,6 +477,141 @@ impl Handle {
+                        .message_mut()
+                        .nlas
+                        .push(Nla::SuppressPrefixLen(rule.suppressPrefixlen as u32));
+                }
+                if rule.suppressIfgroup >= 0 {
+                    request
+                        .message_mut()
+                        .nlas
+                        .push(Nla::SuppressIfGroup(rule.suppressIfgroup as u32));
+                }
+            }
+
+            let table = if rule.table > 0 && rule.table < 256 {
+                rule.table as u8
+            } else {
+                packet::constants::RT_TABLE_UNSPEC
+            };
+
+            // table will be override by Nla::Table
+            request = request.table(table);
+
+            if rule.get_family() == IPFamily::v6 {
+                let mut request = request.v6();
+
+                if !rule.src.is_empty() {
+                    let network = Ipv6Network::from_str(&rule.src)?;
+                    if network.prefix() > 0 {
+                        request = request.source_prefix(network.ip(), network.prefix());
+                    } else {
+                        request
+                            .message_mut()
+                            .nlas
+                            .push(Nla::Source(network.ip().octets().to_vec()));
+                    }
+                }
+
+                if let Err(rtnetlink::Error::NetlinkError(message)) = request.execute().await {
+                    if Errno::from_i32(message.code.abs()) != Errno::EEXIST {
+                        return Err(anyhow!(
+                            "Failed to add IP v6 rule (src: {} table: {} Err: {})",
+                            rule.get_src(),
+                            rule.get_table(),
+                            message
+                        ));
+                    }
+                }
+            } else {
+                let mut request = request.v4();
+                if !rule.src.is_empty() {
+                    let network = Ipv4Network::from_str(&rule.src)?;
+                    if network.prefix() > 0 {
+                        request = request.source_prefix(network.ip(), network.prefix());
+                    } else {
+                        request
+                            .message_mut()
+                            .nlas
+                            .push(Nla::Source(network.ip().octets().to_vec()));
+                    }
+                }
+
+                if let Err(rtnetlink::Error::NetlinkError(message)) = request.execute().await {
+                    if Errno::from_i32(message.code.abs()) != Errno::EEXIST {
+                        return Err(anyhow!(
+                            "Failed to add IP v4 rule (src: {} table: {} Err: {})",
+                            rule.get_src(),
+                            rule.get_table(),
+                            message
+                        ));
+                    }
+                }
+            }
+        }
```

Comment:
Thx！
So many if else or nested, please optimize the code and reduce excessive use of if-else statements.

Reply:
I have replaced the outermost if-else with match but it is not ideal.   Could you provide some suggestions for further optimization?